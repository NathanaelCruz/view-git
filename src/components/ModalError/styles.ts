import styled from 'styled-components'

export const ContainerModal = styled.div`
  background-color: rgba(0, 0, 0, 0.3);
  position: absolute;
  top: 0;
  right: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 10;
`

export const ModalError = styled.div`
  width: 100%;
  max-width: 42rem;
  height: 15rem;
  padding: ${(props) => props.theme.measures.small};
  background-color: ${(props) => props.theme.colors.white};
  border-radius: 5px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
`

export const ModalErrorHeader = styled.header`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1rem 0;
`

export const ModalErrorTitle = styled.h3`
  font: 600
    ${(props) => props.theme.measures.medium + ` ` + props.theme.font.primary};
`

export const ModalErrorClose = styled.button`
  border: none;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0.5rem;
  color: ${(props) => props.theme.colors.white};
  background-color: ${(props) => props.theme.colors.black};
  transition: 0.2s ease-in-out;
  cursor: pointer;

  &:hover {
    background-color: ${(props) => props.theme.colors.primary};
  }
`

export const ModalErrorBody = styled.p`
  line-height: 1.5;
`
