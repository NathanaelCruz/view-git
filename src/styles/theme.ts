const theme = {
  colors: {
    background: '#FFFAFA',
    white: '#FFFAFA',
    black: '#24282E',
    grayDark: '#1C1F23',
    primary: '#ed5140',
    heading: '#ed5140',
    gold: '#f1c40f',
    blue: '#2980b9',
    boxShadow: 'rgba(165, 165, 165, 0.56)',
    text: '#3A3A3A'
  },
  measures: {
    bigSmall: '0.9rem',
    small: '1.3rem',
    medium: '2rem',
    large: '3rem',
    bigLarge: '4.5rem'
  },
  font: {
    primary: `'Open Sans', sans-serif`,
    heading: `'Lobster', cursive`
  }
}

export default theme
