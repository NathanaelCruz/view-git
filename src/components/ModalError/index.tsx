import React, { useContext } from 'react'
import { Close } from '@styled-icons/material'
import * as S from './styles'
import { ViewGitContext } from '../../contexts/ViewGitContext'

type ModalErrorProps = {
  message: string
}

const ModalError: React.FC<ModalErrorProps> = ({ message }) => {
  const { isOpenModalError } = useContext(ViewGitContext)

  return (
    <S.ContainerModal role="dialog">
      <S.ModalError role="document">
        <S.ModalErrorHeader>
          <S.ModalErrorTitle>Ops! Algo está errado</S.ModalErrorTitle>
          <S.ModalErrorClose
            type="button"
            onClick={() => isOpenModalError('')}
            data-testid="button-close-modal"
          >
            <Close size={24} title="Ícone de X" />
          </S.ModalErrorClose>
        </S.ModalErrorHeader>
        <S.ModalErrorBody>{message}</S.ModalErrorBody>
      </S.ModalError>
    </S.ContainerModal>
  )
}

export default ModalError
