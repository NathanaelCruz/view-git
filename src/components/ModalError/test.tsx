import { render, screen } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from '../../styles/theme'

import ModalError from './index'

describe('Modal Error', () => {
  it('Deverá renderizar o modal de error', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <ModalError message="Preencha um campo válido" />
      </ThemeProvider>
    )

    expect(container).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('Deverá renderizar um role dialog', () => {
    render(
      <ThemeProvider theme={theme}>
        <ModalError message="Preencha um campo válido" />
      </ThemeProvider>
    )
    expect(screen.getByRole('dialog')).toBeInTheDocument()
  })

  it('Deverá renderizar um role document', () => {
    render(
      <ThemeProvider theme={theme}>
        <ModalError message="Preencha um campo válido" />
      </ThemeProvider>
    )
    expect(screen.getByRole('document')).toBeInTheDocument()
  })

  it('Deverá renderizar uma message', () => {
    render(
      <ThemeProvider theme={theme}>
        <ModalError message="Preencha um campo válido" />
      </ThemeProvider>
    )
    expect(screen.getByText(/preencha um campo válido/i)).toBeInTheDocument()
  })
})
