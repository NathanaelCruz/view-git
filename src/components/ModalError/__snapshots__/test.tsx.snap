// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Error Deverá renderizar o modal de error 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  background-color: rgba(0,0,0,0.3);
  position: absolute;
  top: 0;
  right: 0;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 10;
}

.c1 {
  width: 100%;
  max-width: 42rem;
  height: 15rem;
  padding: 1.3rem;
  background-color: #FFFAFA;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1rem 0;
}

.c3 {
  font: 600 2rem 'Open Sans',sans-serif;
}

.c4 {
  border: none;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.5rem;
  color: #FFFAFA;
  background-color: #24282E;
  -webkit-transition: 0.2s ease-in-out;
  transition: 0.2s ease-in-out;
  cursor: pointer;
}

.c4:hover {
  background-color: #ed5140;
}

.c6 {
  line-height: 1.5;
}

<div
  class="c0"
  role="dialog"
>
  <div
    class="c1"
    role="document"
  >
    <header
      class="c2"
    >
      <h3
        class="c3"
      >
        Ops! Algo está errado
      </h3>
      <button
        class="c4"
        data-testid="button-close-modal"
        type="button"
      >
        <svg
          class="c5"
          fill="currentColor"
          focusable="false"
          height="24"
          role="img"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Ícone de X
          </title>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </header>
    <p
      class="c6"
    >
      Preencha um campo válido
    </p>
  </div>
</div>
`;
